node/tools/src/k8s.rs

```diff
@@ -22,22 +25,38 @@ pub async fn get_client() -> anyhow::Result<Client> {
 }
 
 /// Get a kube client
-pub async fn get_consensus_node_ips(client: &Client) -> anyhow::Result<Vec<String>> {
+pub async fn get_consensus_node_ips(client: &Client) -> anyhow::Result<Vec<SocketAddr>> {
     let pods: Api<Pod> = Api::namespaced(client.clone(), "consensus");
     let lp = ListParams::default();
     let pod = pods.list(&lp).await?;
-    let a: Vec<String> = pod
+    let a: Result<Vec<SocketAddr>, _> = pod
         .into_iter()
         .filter(|pod| {
-            pod.clone()
-                .metadata
-                .name
-                .unwrap()
-                .starts_with("consensus-node")
+            if let Some(pod) = pod.clone().metadata.name {
+                pod.contains("consensus-node")
+            } else {
+                false
+            }
+        })
+        .map(|pod| {
+            let pod_ip = pod.status.and_then(|status: PodStatus| status.pod_ip);
```

Comment:
argument type annotations in these closures shouldn't be necessary, no?

Reply:
Right, I already removed them